cmd/daemon/player.go

```diff
@@ -259,6 +282,36 @@ func (p *AppPlayer) handlePlayerCommand(req dealer.RequestPayload) error {
 	case "skip_prev":
 		return p.skipPrev()
 	case "skip_next":
+		// todo: refactor
+		if req.Command.Track != nil {
+			ctxTracks := p.state.tracks
+
+			err := ctxTracks.TrySeek(
+				func(track *connectpb.ContextTrack) bool {
+					return track.Uri == req.Command.Track.Uri
+				},
+			)
+			if err != nil {
+				return err
+			}
+
+			p.state.player.Timestamp = time.Now().UnixMilli()
+			p.state.player.PositionAsOfTimestamp = 0
+
+			p.state.tracks = ctxTracks
+			p.state.player.Track = ctxTracks.CurrentTrack()
+			p.state.player.PrevTracks = ctxTracks.PrevTracks()
+			p.state.player.NextTracks = ctxTracks.NextTracks()
+			p.state.player.Index = ctxTracks.Index()
+
+			err = p.loadCurrentTrack(p.state.player.IsPaused)
+			if err != nil {
+				return err
+			}
+
+			return nil
+
+		}
```

Comment:
We should try to reuse as much code as possible, specifically when calling `loadCurrentTrack`. That said, I am not sure what would be the best approach since there are already multiple places where this is not done.

The seeking can be done as follows:
```go
contextSpotType := librespot.InferSpotifyIdTypeFromContextUri(p.state.player.ContextUri)
if err := ctxTracks.TrySeek(tracks.ContextTrackComparator(contextSpotType, req.Command.Track)); err != nil {
    return err
}
```

Reply:
What are your wishes then? Should I leave the p.state. ... assignments like they are currently, or should i try to refactor it myself?

Reply:
I think it can stay this way for now, we'll refactor it eventually. Not sure why you are extracting `p.state.tracks` and reassigning it.

Reply:
Fair point. I think it should be ready to merge now.